### PR TITLE
Add Documentation for Cloning a Linked List with Random and Next Pointers

### LINKED ISSUE #1528: [Idea]: Add Documentation for Cloning a Linked List with Random and Next Pointers
### Idea Title

Add Documentation for Cloning a Linked List with Random and Next Pointers

### Idea Description

I would like to request the addition of detailed documentation that explains the process of cloning a linked list with both random and next pointers. This topic is crucial for understanding how to handle complex linked list structures in our codebase.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [ ] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by riyaa060**
### Description
This pull request introduces detailed documentation for cloning a linked list that includes both random and next pointers. The new documentation aims to clarify the process and provide a comprehensive guide for developers.

Fixes #1528 

### Type of change
- [x] Added new document

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/DSA-Problem-Solution/flattening-a-linked-list.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Flattening a Linked List
+
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/flattening-a-linked-list.md]: ⁉️

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Write your code based on our format

**Comment by riyaa060** [comment]
Done @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**